plugins/org.eclipse.xtend.core/src/org/eclipse/xtend/core/macro/ConstantExpressionsInterpreter.xtend

```diff
@@ -365,9 +365,11 @@ class ConstantExpressionsInterpreter extends AbstractConstantExpressionsInterpre
 	}
 	
 	protected def evaluateField(XAbstractFeatureCall call, JvmField field, Context context) {
+		if (field.isSetConstant) {
```

Comment:
isSetConstant only indicates that the boolean flag 'isConstant' was set. You'll have to check field.isConstant, too.